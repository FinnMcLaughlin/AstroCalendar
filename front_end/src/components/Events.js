import React, { Component } from 'react';
import EventObject from './EventObject.js';
import PropTypes from 'prop-types';

class Events extends Component {    
    render(){
        return (
            <div style={eventsDivStyle}>
                {this.props.event_list.map((event) =>
                    <EventObject  key={event.id} event_info={event} setReminder={this.props.setReminder} />
                )}
            </div>
        )
    }
}

const eventsDivStyle = {
    backgroundColor: '#090428',
    width: '50%',
    margin: 'auto',
}

// PropTypes
Events.propTypes = {
    event_list: PropTypes.array.isRequired,
    setReminder: PropTypes.func.isRequired
}

export default Events;