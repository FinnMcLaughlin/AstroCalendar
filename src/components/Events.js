import React, { Component } from 'react';
import EventObject from './EventObject.js';
import PropTypes from 'prop-types';

class Events extends Component {    
    render(){
        return this.props.event_list.map((event) =>
            <EventObject key={event.id} event_info={event} setReminder={this.props.setReminder} />
        );
    }
}


// PropTypes
Events.propTypes = {
    event_list: PropTypes.array.isRequired,
    setReminder: PropTypes.func.isRequired
}

export default Events;