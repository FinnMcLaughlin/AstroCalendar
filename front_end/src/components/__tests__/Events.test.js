import '@testing-library/jest-dom';
import { render, cleanup } from '@testing-library/react'
import Events from '../Events'

afterEach(() => {
    cleanup()
})

const test_events_array = [
    {
        id: 1,
        event_name: "St. Patricks Day",
        event_date: "17/03/2021",
        reminder_set: true
    },
    {
        id: 2,
        event_name: "Halloween",
        event_date: "31/10/2021",
        reminder_set: false
    }
]

function setReminder(event_id){
    console.log("Reminder set for event: ", event_id)
}

test("renders all events correctly", () => {
    const { getAllByTestId } = render(<Events event_list={test_events_array} setReminder={setReminder}/>)

    const all_events = getAllByTestId("event-obj-div")

    expect(all_events[0].textContent).toContain("St. Patricks Day")
    expect(all_events[1].textContent).toContain("Halloween")

    expect(all_events).toMatchSnapshot()
})