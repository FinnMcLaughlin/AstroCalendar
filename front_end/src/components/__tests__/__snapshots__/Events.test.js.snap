// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all events correctly 1`] = `
Array [
  <div
    data-testid="event-obj-div"
    style="background-color: rgb(219, 22, 22);"
  >
    <h2>
      St. Patricks Day
    </h2>
    <h3>
      17/03/2021
    </h3>
    <button>
      Create Reminder
    </button>
  </div>,
  <div
    data-testid="event-obj-div"
    style="background-color: rgb(244, 244, 244);"
  >
    <h2>
      Halloween
    </h2>
    <h3>
      31/10/2021
    </h3>
    <button>
      Create Reminder
    </button>
  </div>,
]
`;
